File: scripts/cleanup_unanalyzed.js
const { createClient } = require('@supabase/supabase-js')
require('dotenv').config({ path: 'taxonomy-app/.env.local' })

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL
const supabaseServiceKey = process.env.SUPABASE_SERVICE_ROLE_KEY // ต้องใช้ Service Role เพื่อลบข้อมูล

if (!supabaseUrl || !supabaseServiceKey) {
  console.error('Missing Supabase credentials in .env.local')
  process.exit(1)
}

const supabase = createClient(supabaseUrl, supabaseServiceKey)

async function cleanupUnanalyzedProducts() {
  console.log('🚀 เริ่มการลบสินค้าที่ยังไม่ผ่านการวิเคราะห์ (1,620 รายการ)...')
  
  // ลบข้อมูล
  const { data, error, count } = await supabase
    .from('products')
    .delete({ count: 'exact' })
    .or('confidence_score.eq.0,embedding.is.null')

  if (error) {
    console.error('❌ เกิดข้อผิดพลาดในการลบ:', error.message)
  } else {
    console.log(`✅ ลบสำเร็จทั้งหมด: ${count} รายการ`)
  }
}

cleanupUnanalyzedProducts()

File: scripts/cleanup_unanalyzed.py
import os
from supabase import create_client, Client
from dotenv import load_dotenv

# โหลด config
load_dotenv('taxonomy-app/.env.local')

url = os.environ.get("NEXT_PUBLIC_SUPABASE_URL")
key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY") # ใช้ Service Role สำหรับลบ

if not url or not key:
    print("❌ ไม่พบ Supabase credentials ใน .env.local")
    exit(1)

supabase: Client = create_client(url, key)

def cleanup():
    print("🚀 เริ่มการลบสินค้าที่ยังไม่ผ่านการวิเคราะห์ (1,620 รายการ)...")
    
    # ดำเนินการลบ
    # เงื่อนไข: confidence_score == 0 หรือ embedding IS NULL
    try:
        response = supabase.table("products") \
            .delete() \
            .or_("confidence_score.eq.0,embedding.is.null") \
            .execute()
        
        count = len(response.data)
        print(f"✅ ลบสำเร็จทั้งหมด: {count} รายการ")
    except Exception as e:
        print(f"❌ เกิดข้อผิดพลาด: {str(e)}")

if __name__ == "__main__":
    cleanup()
